appstudio-controller/controllers/appstudio.redhat.com/environment_controller.go

```diff
@@ -78,94 +82,127 @@ func (r *EnvironmentReconciler) Reconcile(ctx context.Context, req ctrl.Request)
+	if existingEnvironment == nil {
+		log.Info("Creating GitOpsDeploymentManagedEnvironment", "managedEnv", managedEnvironment.Name)
 
-			// Success: the resource has been created.
+		if err := r.Client.Create(ctx, managedEnvironment); err != nil {
+			return ctrl.Result{}, fmt.Errorf("unable to create new GitOpsDeploymentManagedEnvironment: %v", err)
+		}
+		sharedutil.LogAPIResourceChangeEvent(managedEnvironment.Namespace, managedEnvironment.Name, managedEnvironment, sharedutil.ResourceCreated, log)
+	} else {
+		if reflect.DeepEqual(existingEnvironment.Spec, managedEnvironment.Spec) {
+			// If the spec field is the same, no more work is needed.
 			return ctrl.Result{}, nil
-
-		} else {
-			// For any other error, return it
-			return ctrl.Result{}, fmt.Errorf("unable to retrieve existing GitOpsDeploymentManagedEnvironment '%s': %v",
-				currentManagedEnv.Name, err)
 		}
-	}
-
-	// C) The GitOpsDeploymentManagedEnvironment already exists, so compare it with the desired state, and update it if different.
-	if reflect.DeepEqual(currentManagedEnv.Spec, desiredManagedEnv.Spec) {
-		// If the spec field is the same, no more work is needed.
-		return ctrl.Result{}, nil
-	}
 
-	log.Info("Updating GitOpsDeploymentManagedEnvironment as a change was detected", "managedEnv", desiredManagedEnv.Name)
+		log.Info("Updating GitOpsDeploymentManagedEnvironment as a change was detected", "managedEnv", managedEnvironment.Name)
 
-	// Update the current object to the desired state
-	currentManagedEnv.Spec = desiredManagedEnv.Spec
+		// Update the current object to the desired state
+		existingEnvironment.Spec = managedEnvironment.Spec
 
-	if err := r.Client.Update(ctx, &currentManagedEnv); err != nil {
-		return ctrl.Result{},
-			fmt.Errorf("unable to update existing GitOpsDeploymentManagedEnvironment '%s': %v", currentManagedEnv.Name, err)
+		if err := r.Client.Update(ctx, existingEnvironment); err != nil {
+			return ctrl.Result{},
+				fmt.Errorf("unable to update existing GitOpsDeploymentManagedEnvironment '%s': %v", existingEnvironment.Name, err)
+		}
+		sharedutil.LogAPIResourceChangeEvent(existingEnvironment.Namespace, existingEnvironment.Name, existingEnvironment, sharedutil.ResourceModified, log)
 	}
-	sharedutil.LogAPIResourceChangeEvent(currentManagedEnv.Namespace, currentManagedEnv.Name, currentManagedEnv, sharedutil.ResourceModified, log)
 
 	return ctrl.Result{}, nil
 }
 
-// TODO: GITOPSRVCE-182: Uncomment this code and use it to implement conditions.
-
-// const EnvironmentStatus_ConditionErrorOccurred = "ErrorOccurred"
-
-// func updateStatusConditions(err error) error {
+func (r *EnvironmentReconciler) createWorkspace(ctx context.Context, env *appstudioshared.Environment) (*v1beta1.Workspace, error) {
+	workspace, err := r.getSubWorkspaceFromEnvironment(ctx, env)
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
Think you can use `apierr.IsNotFound(err)` here, I believe, from kube api machinery, rather than `"k8s.io/cri-api/pkg/errors"` (unless you are aware of pre-existing best practices that are counter to that)